dimod/discrete/discrete_quadratic_model.py

```diff
@@ -416,6 +419,24 @@ def get_quadratic_case(self, u, u_case, v, v_case):
         return self._cydqm.get_quadratic_case(
             self.variables.index(u), u_case, self.variables.index(v), v_case)
 
+    def add_constraint_as_quadratic(self, terms: Linear, lagrange_multiplier: float,
+                                    constant: float):
+        """
+        Add a linear constraint of the form \\sum_{ik} a_{ik} x_{ik} + C = 0 to dqm object as a quadratic objective.
+        Args:
+            dqm: DiscreteQuadraticModel
+            terms: A list or a generator of tuples of the type (variable, case, bias).
+                Each tuple is evaluated to the term bias * variable_case. All terms in the list/generator are summed.
+            lagrange_multiplier: The coefficient or the penalty strength
+            const: The constant value of the constraint.
+
+        Returns: None. Updates the dqm object
+
+        """
+        variables, cases, biases = zip(*terms)
```

Comment:
IMO we should not accept a generator if we're just going to dump it all into memory in the first line.

Reply:
I am wrong, since even in the case that you're giving it as a list, it's making a copy into the three different lists. I suppose a generator saves on half the memory in this case.

Reply:
```
for v, c, b in terms:
    variables.append(self.variables.index(v))
    cases.append(c)
    biases.append(b)
```
will save a decent chunk of memory without a performance hit I don't think.